test/e2e/e2e_test.go

```diff
@@ -42,35 +43,70 @@ import (
+		kubeCli                = "kubectl"
+		ocpCli                 = "oc"
+		dockerBin              = "docker"
+		podmanBin              = "podman"
+		deployEmulated         = "deploy-emulated"
+		deployOCPEmulated      = "ocp-deploy-emulated"
+		clientBin              = kubeCli
+		containerTool          = dockerBin
+		deployEmulatedArg      = deployEmulated
+	)
+	ocpMode := os.Getenv("OCP_MODE")
+	if ocpMode != "" {
+		isRunningOnOCP = true
+		clientBin = ocpCli
+		containerTool = podmanBin
+		deployEmulatedArg = deployOCPEmulated
+	}
 
 	BeforeAll(func() {
-		fmt.Println("Setting up Kind cluster")
-		cmd := exec.Command("kind", "create", "cluster", "--image", "kindest/node:v1.30.0@sha256:047357ac0cfea04663786a612ba1eaba9702bef25227a794b52890dd8bcd692e")
-		output, err := cmd.CombinedOutput()
-		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to create Kind cluster: %s", output))
+		if !isRunningOnOCP {
+			fmt.Println("Setting up Kind cluster")
+			cmd := exec.Command("kind", "create", "cluster", "--image", "kindest/node:v1.30.0@sha256:047357ac0cfea04663786a612ba1eaba9702bef25227a794b52890dd8bcd692e")
+			output, err := cmd.CombinedOutput()
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to create Kind cluster: %s", output))
+		}
 
 		By("creating manager namespace")
-		cmdNamespace := exec.Command("kubectl", "create", "ns", namespace)
+		cmdNamespace := exec.Command(clientBin, "create", "ns", namespace)
 		outputNs, err := cmdNamespace.CombinedOutput()
 		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to create namespace: %s", outputNs))
 
 		By("installing cert manager")
-		cmdCm := exec.Command("kubectl", "apply", "-f", "https://github.com/cert-manager/cert-manager/releases/download/v1.15.3/cert-manager.yaml")
+		cmdCm := exec.Command(clientBin, "apply", "-f", "https://github.com/cert-manager/cert-manager/releases/download/v1.15.3/cert-manager.yaml")
 		outputCm, err := cmdCm.CombinedOutput()
 		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to deploy cert manager: %s", outputCm))
 
 		By("validating the cert-manager-webhook pod to be ready as expected")
-		EventuallyWithOffset(1, isResourceReady, 2*time.Minute, time.Second).WithArguments("pod", "app=webhook", "cert-manager").Should(BeTrue())
+		EventuallyWithOffset(1, isResourceReady, 2*time.Minute, time.Second).WithArguments(clientBin, "pod", "app=webhook", "cert-manager").Should(BeTrue())
 	})
 
 	AfterAll(func() {
-		fmt.Println("Deleting the cluster")
-		cmd := exec.Command("kind", "delete", "cluster")
-		output, err := cmd.CombinedOutput()
-		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to create Kind cluster: %s", output))
+		if !isRunningOnOCP {
+			fmt.Println("Deleting the cluster")
+			cmd := exec.Command("kind", "delete", "cluster")
+			output, err := cmd.CombinedOutput()
+			Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to create Kind cluster: %s", output))
+		} else {
+			cmd := exec.Command("make", "ocp-undeploy-emulated")
```

Comment:
we don't need the logic to run make within the e2e. Was there a conversation to keep it in here?